### PR TITLE
Separate Lotus (Postgres handler) from OSMCache

### PR DESCRIPTION
**Pull request by GuillaumeLescure**
* allow to have a null Lotus, in that case, requestes are not sent
* minor cosmetic

### CODE REVIEW

File: source/ed/osm2ed.cpp [hunk 1]
```diff
@@ -1131,7 +1164,7 @@ int osm2ed(int argc, const char** argv) {
     persistor.clean_georef();
     persistor.clean_poi();
 
-    ed::connectors::OSMCache cache(connection_string, cities_cnx);
+    ed::connectors::OSMCache cache(std::unique_ptr<Lotus>(new Lotus(connection_string)), cities_cnx);
```
Comment (kinnou02) [inline source/ed/osm2ed.cpp]:
```suggestion
    ed::connectors::OSMCache cache(std::make_unique<Lotus>(connection_string), cities_cnx);
```

**Comment by GuillaumeLescure** [comment]
Test this please

**Review by nonifier — APPROVED** [review]

**Comment by GuillaumeLescure** [comment]
Test this please.

**Review by benoit-bst — APPROVED** [review]

**Comment by benoit-bst** [comment]
Jenkins, test this please !

**Comment by GuillaumeLescure** [comment]
Build is ok, hook with Jenkins is broken.
See : https://ci.navitia.io/view/navitia/job/custom_navitia_pull_request_build/98/

**Review by nonifier — APPROVED** [review]

**Review by kinnou02 — APPROVED** [review]

**Comment by kinnou02** [comment]
Jenkins, test this please.

**Pull request merged**